python/mxnet/symbol/symbol.py

```diff
@@ -1470,6 +1470,10 @@ def optimize_for(self, backend, args=None, aux=None, ctx=None,
         ctx : Context, optional
             Device context, used to infer stypes
 
+        is_np_sym : boolean, optional
+            Output symbol type
+            - If true, output type is np symbol, otherwise nd symbol.
+
```

Comment:
nd symbol will be removed soon anyways. Can you rely on the global `is_np` state (and if needed adapt the state in the `build_cache` function)

Reply:
Will be changing it